ada/language/ast.py

```diff
@@ -1317,8 +1317,7 @@ def base_subp_declarations():
         return Entity.is_subprogram.then(
             lambda _: Entity.subp_spec_or_null.then(
                 lambda spec: spec.primitive_subp_of.find(
-                    lambda t:
```

Comment:
You have another opportunity for simplification here:

https://github.com/AdaCore/libadalang/blob/master/ada/language/ast.py#L942

Reply:
Actually I already implemented it (see https://github.com/AdaCore/libadalang/pull/292/commits/6620cfaa398c208171c49f76d99ee99812725e11) and I saw this opportunity but I then noticed that the semantics of the code you linked is not exactly the same as what I factored out. However if you think this change in semantics is okay, I can still proceed with the change.

Reply:
Ok fair enough